### PR TITLE
docs: fix cargo doc errors and warnings

### PR DESCRIPTION
**Pull request by evenyag**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?
This PR fixes errors and warnings while running
```
cargo doc --workspace --no-deps --document-private-items
```

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Comment by killme2008** [comment]
Can we add this checking to CI?

File: src/common/function-macro/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,226 @@
+// Copyright 2023 Greptime Team
```
Comment (killme2008) [inline src/common/function-macro/src/lib.rs:1]: So we deleted this file before?

**Review by killme2008 — APPROVED** [review]

File: src/common/function-macro/src/lib.rs
Comment (WenyXu) [inline src/common/function-macro/src/lib.rs:1]: According to https://github.com/GreptimeTeam/greptimedb/pull/2418, Let's rebase the develop.

File: src/common/function-macro/src/lib.rs (hunk 1 above)
Comment (waynexia) [reply src/common/function-macro/src/lib.rs:1]: I renamed it to `common/macro` in #2418

**Comment by evenyag** [comment]
@WenyXu PTAL

**Review by WenyXu — APPROVED** [review]

**Pull request merged**